app/src/main/java/com/github/featuredetectandroid/utils/PhotoAnalyzer.kt

```diff
@@ -46,7 +47,9 @@ class PhotoAnalyzer(private val outputViewModel: OutputViewModel) : ImageAnalysi
             outputViewModel.featureDetector?.let { detector ->
                 if (detector.width != width) detector.width = width
                 if (detector.height != height) detector.height = height
+                val startTime = SystemClock.elapsedRealtime()
                 val (keypoints, _) = detector.detect(luminanceArrayToRGB(oriented))
+                outputViewModel.calcTimeMs = SystemClock.elapsedRealtime() - startTime
```

Comment:
So, here we also measure the time `luminanceArrayToRGB` take which we don't want to. Save save it to `rgbArray` and pass it as an argument.

Reply:
fixed